src/main/kotlin/no/nav/familie/ks/sak/common/util/Tid.kt

```diff
@@ -18,4 +24,94 @@ fun YearMonth.tilKortString() = this.format(DateTimeFormatter.ofPattern("MM.yy",
 fun YearMonth.tilMånedÅr() = this.format(DateTimeFormatter.ofPattern("MMMM yyyy", nbLocale))
 fun YearMonth.toLocalDate() = LocalDate.of(this.year, this.month, 1)
 
+fun LocalDate.toYearMonth() = YearMonth.from(this)
+fun YearMonth.førsteDagIInneværendeMåned() = this.atDay(1)
+fun YearMonth.sisteDagIInneværendeMåned() = this.atEndOfMonth()
+
+fun YearMonth.erSammeEllerTidligere(toCompare: YearMonth): Boolean = this.isBefore(toCompare) || this == toCompare
+
+fun inneværendeMåned(): YearMonth = LocalDate.now().toYearMonth()
+
+fun LocalDate.nesteMåned(): YearMonth = this.toYearMonth().plusMonths(1)
+
+fun YearMonth.nesteMåned(): YearMonth = this.plusMonths(1)
+
+fun LocalDate.erDagenFør(other: LocalDate?) = other != null && this.plusDays(1).equals(other)
+
+data class Periode(val fom: LocalDate, val tom: LocalDate)
+
+fun LocalDate.erSammeEllerFør(toCompare: LocalDate): Boolean = this.isBefore(toCompare) || this == toCompare
+fun LocalDate.erSammeEllerEtter(toCompare: LocalDate): Boolean = this.isAfter(toCompare) || this == toCompare
+fun LocalDate.erMellom(toCompare: Periode): Boolean = this.erSammeEllerEtter(toCompare.fom) &&
+    this.erSammeEllerEtter(toCompare.tom)
+
+fun LocalDate.førsteDagIInneværendeMåned() = this.withDayOfMonth(1)
+
+fun Periode.overlapperHeltEllerDelvisMed(annenPeriode: Periode) =
+    this.fom.erMellom(annenPeriode) ||
+        this.tom.erMellom(annenPeriode) ||
+        annenPeriode.fom.erMellom(this) ||
+        annenPeriode.tom.erMellom(this)
+
+fun Periode.tilMånedPeriode(): MånedPeriode = MånedPeriode(fom = this.fom.toYearMonth(), tom = this.tom.toYearMonth())
+
 data class MånedPeriode(val fom: YearMonth, val tom: YearMonth)
+
+fun MånedPeriode.inkluderer(yearMonth: YearMonth) = yearMonth >= this.fom && yearMonth <= this.tom
+fun MånedPeriode.overlapperHeltEllerDelvisMed(annenPeriode: MånedPeriode) =
+    this.inkluderer(annenPeriode.fom) ||
+        this.inkluderer(annenPeriode.tom) ||
+        annenPeriode.inkluderer(this.fom) ||
+        annenPeriode.inkluderer(this.tom)
+
+fun MånedPeriode.erMellom(annenPeriode: MånedPeriode) =
+    annenPeriode.inkluderer(this.fom) && annenPeriode.inkluderer(this.tom)
+
+fun LocalDate.førsteDagINesteMåned() = this.plusMonths(1).withDayOfMonth(1)
+
+fun erBack2BackIMånedsskifte(tilOgMed: LocalDate?, fraOgMed: LocalDate?): Boolean =
+    tilOgMed?.erDagenFør(fraOgMed) == true && tilOgMed.toYearMonth() != fraOgMed?.toYearMonth()
+
+fun DatoIntervallEntitet.erInnenfor(dato: LocalDate): Boolean =
+    when {
+        fom == null && tom == null -> true
+        fom == null -> dato.erSammeEllerFør(tom!!)
+        tom == null -> dato.erSammeEllerEtter(fom)
+        else -> dato.erSammeEllerEtter(fom) && dato.erSammeEllerFør(tom)
+    }
+
+fun slåSammenOverlappendePerioder(overlappendePerioder: Collection<DatoIntervallEntitet>): List<DatoIntervallEntitet> {
```

Comment:
Synes denne metoden var vanskelig å forstå.
Blir nesten som en blackbox, man vet bare at resultatet er at periodene slås sammen 😛 

Den er uansett tatt fra ba-sak, så det går fint foreløpig. Vi burde ta en prat med @halvorbmundal for å se om den kan erstattes med tidslinjer.

Reply:
Jepp, men som diskutert kan jeg se på det etter Halvor sin PR.